packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -21,89 +22,115 @@ import {
   buildSearchParameterSerializer,
   Unsubscribe,
 } from '@coveo/headless';
-import {InitializeEvent} from '../../utils/initialization-utils';
+import {
+  InterfaceContext,
+  InitializeEvent,
+} from '../../utils/initialization-utils';
+import i18next, {i18n} from 'i18next';
+import Backend, {BackendOptions} from 'i18next-http-backend';
+
+export type InitializationOptions = Pick<
+  HeadlessConfigurationOptions,
+  'accessToken' | 'organizationId' | 'renewAccessToken' | 'platformUrl'
+>;
 
 @Component({
   tag: 'atomic-search-interface',
   shadow: true,
+  assetsDirs: ['lang'],
 })
 export class AtomicSearchInterface {
   @Element() host!: HTMLDivElement;
-  @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'info';
+  @Prop() logLevel?: LogLevel;
+  @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({reflect: true}) language = 'en'; // TODO: make watchable and update i18next language on change
+  @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
-  @State() engine?: Engine;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
   private initialized = false;
 
-  componentWillLoad() {
-    if (this.sample) {
-      this.initialize(HeadlessEngine.getSampleConfiguration());
-    }
+  private get context(): InterfaceContext {
+    return {engine: this.engine!, i18n: this.i18n};
   }
 
-  disconnectedCallback() {
-    this.unsubscribe();
+  @Method() public async initialize(options: InitializationOptions) {
+    if (this.engine) {
```

Comment:
Should this be using `this.initialized` instead ?

And then, inside `this.initEngine`, we could have this check here.

Reply:
the engine is created right await, this.initialized is set to true after everything is loaded (might be a few ms of difference) it wont stop users from calling initialize successively